src/main/java/namor/template/calculator/RPNcalculator.java

```diff
@@ -0,0 +1,64 @@
+package namor.template.calculator;
+
+import namor.template.stack.ArrayStack;
+
+public class RPNcalculator {
+    private int result;
+    private ArrayStack numbers = new ArrayStack();
+
+    public void isCorrect(String rpnString) throws IllegalArgumentException {
+        if(rpnString == null || rpnString == ""){
+            throw new IllegalArgumentException("Input string is empty!");
+        }
+
+        for(int i = 0; i < rpnString.length(); i++){
+            char x;
+            x = rpnString.charAt(i);
+
+            if(!(Character.isDigit(x) || x == '/' || x == '*' ||
+                    x == '+'|| x == '-' || x == ' ')){
+                throw new IllegalArgumentException("Input elements are wrong!");
+            }
+        }
+    }
+
+    public int calculate(String rpnString){
+        int firstNumber;
+        int secondNumber;
+
+        isCorrect(rpnString);
+
+        for (String retval: rpnString.split(" ")) {
+            if(retval.matches("^[0-9]+$")){
+                numbers.push(Integer.parseInt(retval));
+            } else {
+
+                if(Signs.ADDITION.getKey().equals(retval)){
```

Comment:
You can use Switch operator instead if-else to avoid such a massy things as 'Signs.ADDITION.getKey().equals(retval)'